### PR TITLE
initial commit ebsview sql

### PR DESCRIPTION
**Pull request by tsengtammy**
sql to generate views for ebsview_signal_report, signal_verification, and signal_investigation

### CODE REVIEW

File: analytics/ebs/ebsview_signal_investigation.sql [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        (doc #>> '{fields, scdsc_investigation, no_suspected_cases}')::int AS num_suspected_cases,
+        (doc #>> '{fields, scdsc_investigation, no_human_cases_hospitalized}')::int AS num_human_cases_hospitalized,
+        (doc #>> '{fields, scdsc_investigation, no_human_deaths}')::int AS num_human_deaths,
+        (doc #>> '{fields, scdsc_investigation, event_risk_classification}')::text AS event_risk_classification,
+        (doc #>> '{fields, scdsc_investigation, response_recommendations}')::text AS response_recommendations
+    FROM
+        couchdb
+    WHERE
+        doc ->> 'form' = 'scdsc_investigation'
+);
```
Comment (yrimal) [inline analytics/ebs/ebsview_signal_investigation.sql:48]: Don't forget to change the owner of the object to `full_access` user.

File: analytics/ebs/ebsview_signal_verification.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        (doc #>> '{fields, signal_verification, additional_details}')::text as additional_details,
+        (doc #>> '{fields, signal_verification, source}')::text as signal_source,
+        (doc #>> '{fields, signal_verification, source}')::text as source_specify,
+        (doc #>> '{fields, signal_verification, threat_exists}')::text AS threat_exists,
+        (doc #>> '{fields, signal_verification, verification_date}')::date as verification_date
+    FROM
+        couchdb
+    WHERE
+        doc ->> 'form' = 'cha_signal_verification'
+);
```
Comment (yrimal) [inline analytics/ebs/ebsview_signal_verification.sql:26]: We will need to change owner to full_access. Something like : 
`ALTER VIEW chtview_signal_report OWNER TO full_access;`

File: analytics/ebs/ebsview_signal_investigation.sql [hunk 3]
```diff
@@ -0,0 +1,48 @@
+CREATE OR REPLACE VIEW chtview_signal_investigation AS
+(
+    SELECT
+        doc ->> '_id' AS uuid,
+        doc ->> 'form' AS form,
+        doc #>> '{contact,_id}' AS reported_by,
+        doc #>> '{contact,parent,_id}' AS reported_by_parent,
+        doc #>> '{geolocation,latitude}' AS latitude,
```
Comment (michaelkohn) [inline analytics/ebs/ebsview_signal_investigation.sql:8]: I normally like to put lat/long and reported as the last fields in the view.

File: analytics/ebs/ebsview_signal_investigation.sql [hunk 4]
```diff
@@ -0,0 +1,48 @@
+CREATE OR REPLACE VIEW chtview_signal_investigation AS
```
Comment (michaelkohn) [inline analytics/ebs/ebsview_signal_investigation.sql:1]: Please change to ebsview_signal_investigation

File: analytics/ebs/ebsview_signal_report.sql [hunk 5]
```diff
@@ -0,0 +1,15 @@
+CREATE OR REPLACE VIEW chtview_signal_report AS
```
Comment (michaelkohn) [inline analytics/ebs/ebsview_signal_report.sql:1]: Please change to ebsview_signal_report

File: analytics/ebs/ebsview_signal_verification.sql [hunk 6]
```diff
@@ -0,0 +1,26 @@
+CREATE OR REPLACE VIEW chtview_signal_verification AS
```
Comment (michaelkohn) [inline analytics/ebs/ebsview_signal_verification.sql:1]: Please change to ebsview_signal_verification

File: analytics/ebs/ebsview_signal_verification.sql [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+(
+    SELECT
+        doc ->> '_id' AS uuid,
+        doc ->> 'form' AS form,
+        doc #>> '{contact,_id}' AS reported_by,
+        doc #>> '{contact,parent,_id}' AS reported_by_parent,
+        doc #>> '{geolocation,latitude}' AS latitude,
```
Comment (michaelkohn) [inline analytics/ebs/ebsview_signal_verification.sql:8]: Please move lat/long/reported to the end of the view

File: analytics/ebs/ebsview_signal_verification.sql [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        to_timestamp((NULLIF(couchdb.doc ->> 'reported_date'::text, ''::text)::bigint / 1000)::double precision) AS reported,
+        (doc #>> '{fields,signal_id}')::int AS signal_id,
+        (doc #>> '{fields, signal_verification, signal_desc}')::text AS signal_desc,
+        (doc #>> '{fields, signal_verification, signal_type}')::text AS signal_confirmation,
+        (doc #>> '{fields, signal_verification, signal_start_date}')::date as signal_start_date,
+        (doc #>> '{fields, signal_verification, no_people_symptoms}')::int as num_people_symptoms,
+        (doc #>> '{fields, signal_verification, no_people_dead}')::int as num_people_dead,
+        (doc #>> '{fields, signal_verification, additional_details}')::text as additional_details,
+        (doc #>> '{fields, signal_verification, source}')::text as signal_source,
+        (doc #>> '{fields, signal_verification, source}')::text as source_specify,
```
Comment (michaelkohn) [inline analytics/ebs/ebsview_signal_verification.sql:19]: This should be `{fields, signal_verification, source_specify}`

**Comment by tsengtammy** [comment]
Sorry for the late update! Github emails were getting filtered out of my main inbox. Quick q - wanted to confirm `signal_id` should be an int and not a string? I.e. I noticed id's like `00590`, so checking that we do want them to be stored as `590` instead of `"00590"`

**Comment by michaelkohn** [comment]
Good question @tsengtammy.  Historically I've never seen one of these IDs have characters in them in a production environment. I think in the not too distant future we might start seeing them (have chars) though, to differentiate different types of IDs. So ideally we'd store it as text.

**Comment by tsengtammy** [comment]
Thanks @michaelkohn ! I updated the revision to store `signal_id` as text.

**Pull request merged**